src/main/scala/ch/epfl/bluebrain/nexus/admin/projects/Projects.scala

```diff
@@ -0,0 +1,149 @@
+package ch.epfl.bluebrain.nexus.admin.projects
+
+import java.time.Clock
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import cats.Monad
+import cats.effect.ConcurrentEffect
+import cats.syntax.functor._
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig.HttpConfig
+import ch.epfl.bluebrain.nexus.admin.index.Index
+import ch.epfl.bluebrain.nexus.admin.projects.ProjectCommand._
+import ch.epfl.bluebrain.nexus.admin.projects.ProjectRejection._
+import ch.epfl.bluebrain.nexus.admin.projects.ProjectState._
+import ch.epfl.bluebrain.nexus.commons.types.Meta
+import ch.epfl.bluebrain.nexus.commons.types.identity.Identity
+import ch.epfl.bluebrain.nexus.sourcing.Aggregate
+import ch.epfl.bluebrain.nexus.sourcing.akka._
+
+class Projects[F[_]](agg: Agg[F], index: Index)(implicit F: Monad[F], http: HttpConfig, clock: Clock) {
+
+  /**
+    * Creates a project.
+    *
+    * @param label       the project label (segment)
+    * @param description an optional description
+    * @param caller      an implicitly available caller
+    * @return project the created project resource metadata if the operation was successful, a rejection otherwise
+    */
+  def create(label: ProjectLabel, description: Option[String])(implicit caller: Identity): F[ProjectMetaOrRejection] =
+    index.getOrganization(label.organization) match {
+      case Some(org) =>
+        index.getProject(label) match {
+          case None =>
+            val projectId = UUID.randomUUID
+            val command   = CreateProject(projectId, org.id, label, description, Meta(caller, clock.instant))
+            eval(projectId.toString, command)
+          case Some(_) => F.pure(Left(ProjectAlreadyExists))
+        }
+      case None => F.pure(Left(OrganizationDoesNotExist))
+    }
+
+  /**
+    * Updates a project.
+    *
+    * @param label       the project label (segment)
+    * @param description an optional description
+    * @param caller      an implicitly available caller
+    * @return the updated project resource metadata if the operation was successful, a rejection otherwise
+    */
+  def update(label: ProjectLabel, description: Option[String])(implicit caller: Identity): F[ProjectMetaOrRejection] =
+    index.getProject(label) match {
+      case Some(project) =>
+        if (project.deprecated) F.pure(Left(ProjectIsDeprecated))
```

Comment:
This case is handled in `Eval` so we probably don't need this if statement here.